### PR TITLE
[#1484] improvement(*): Add support for listing task dependencies tree.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add gradle task `taskTree` to show task dependencies tree. 

### Why are the changes needed?

For better debugging and developement.

Fix: #1484

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Test locally.  you can use 
```
./gradlew {taskName} taskTree 
```
to show task dependencies tree. Here is an example
```
./gradlew assembleDistribution taskTree
```

The result:

```text
:assembleDistribution
+--- :assembleTrinoConnector
|    \--- :trino-connector:copyLibs
|         +--- :trino-connector:build
|         |    +--- :trino-connector:assemble
|         |    |    \--- :trino-connector:jar
|         |    |         +--- :trino-connector:classes
|         |    |         |    +--- :trino-connector:compileJava
|         |    |         |    |    \--- :clients:client-java-runtime:shadowJar
|         |    |         |    |         +--- :api:jar
|         |    |         |    |         |    +--- :api:classes
|         |    |         |    |         |    |    +--- :api:compileJava
|         |    |         |    |         |    |    \--- :api:processResources
|         |    |         |    |         |    \--- :api:compileJava *
|         |    |         |    |         +--- :common:jar
|         |    |         |    |         |    +--- :common:classes
|         |    |         |    |         |    |    +--- :common:compileJava
|         |    |         |    |         |    |    |    \--- :api:jar *
|         |    |         |    |         |    |    \--- :common:processResources
|         |    |         |    |         |    \--- :common:compileJava *
|         |    |         |    |         +--- :clients:client-java:jar
|         |    |         |    |         |    +--- :clients:client-java:classes
|         |    |         |    |         |    |    +--- :clients:client-java:compileJava
|         |    |         |    |         |    |    |    +--- :api:jar *
|         |    |         |    |         |    |    |    \--- :common:jar *
|         |    |         |    |         |    |    \--- :clients:client-java:processResources
|         |    |         |    |         |    \--- :clients:client-java:compileJava *
|         |    |         |    |         +--- :clients:client-java-runtime:classes
|         |    |         |    |         |    +--- :clients:client-java-runtime:compileJava
|         |    |         |    |         |    |    \--- :clients:client-java:jar *
|         |    |         |    |         |    \--- :clients:client-java-runtime:processResources
|         |    |         |    |         \--- :clients:client-java-runtime:compileJava *
|         |    |         |    \--- :trino-connector:processResources
|         |    |         \--- :trino-connector:compileJava *
|         |    \--- :trino-connector:check
|         |         +--- :trino-connector:spotlessCheck
|         |         |    +--- :trino-connector:spotlessJavaCheck
|         |         |    |    \--- :trino-connector:spotlessJava
|         |         |    |         \--- :spotlessInternalRegisterDependencies
|         |         |    \--- :trino-connector:spotlessKotlinGradleCheck
|         |         |         \--- :trino-connector:spotlessKotlinGradle
|         |         |              \--- :spotlessInternalRegisterDependencies *
|         |         \--- :trino-connector:test
|         |              +--- :trino-connector:classes *
|         |              +--- :trino-connector:compileJava *
|         |              +--- :trino-connector:compileTestJava
|         |              |    +--- :trino-connector:classes *
|         |              |    +--- :trino-connector:compileJava *
|         |              |    \--- :clients:client-java-runtime:shadowJar *
|         |              +--- :trino-connector:testClasses
|         |              |    +--- :trino-connector:compileTestJava *
|         |              |    \--- :trino-connector:processTestResources
|         |              \--- :clients:client-java-runtime:shadowJar *
|         \--- :trino-connector:copyDepends
|              \--- :clients:client-java-runtime:shadowJar *
\--- :compileDistribution
     +--- :copyCatalogLibAndConfigs
     |    +--- :catalogs:catalog-hive:copyLibAndConfig
     |    |    +--- :catalogs:catalog-hive:copyCatalogConfig
     |    |    \--- :catalogs:catalog-hive:copyCatalogLibs
     |    |         +--- :catalogs:catalog-hive:build
     |    |         |    +--- :catalogs:catalog-hive:assemble
     |    |         |    |    \--- :catalogs:catalog-hive:jar
     |    |         |    |         +--- :catalogs:catalog-hive:classes
     |    |         |    |         |    +--- :catalogs:catalog-hive:compileJava
     |    |         |    |         |    |    +--- :api:jar *
     |    |         |    |         |    |    \--- :core:jar
     |    |         |    |         |    |         +--- :core:classes
     |    |         |    |         |    |         |    +--- :core:compileJava
     |    |         |    |         |    |         |    |    +--- :api:jar *
     |    |         |    |         |    |         |    |    +--- :common:jar *
     |    |         |    |         |    |         |    |    \--- :meta:jar
     |    |         |    |         |    |         |    |         +--- :meta:classes
     |    |         |    |         |    |         |    |         |    +--- :meta:compileJava
     |    |         |    |         |    |         |    |         |    |    \--- :meta:generateProto
     |    |         |    |         |    |         |    |         |    |         +--- :meta:extractIncludeProto
     |    |         |    |         |    |         |    |         |    |         \--- :meta:extractProto
     |    |         |    |         |    |         |    |         |    \--- :meta:processResources
     |    |         |    |         |    |         |    |         |         \--- :meta:extractProto *
     |    |         |    |         |    |         |    |         \--- :meta:compileJava *
     |    |         |    |         |    |         |    \--- :core:processResources
     |    |         |    |         |    |         \--- :core:compileJava *
     |    |         |    |         |    \--- :catalogs:catalog-hive:processResources
     |    |         |    |         \--- :catalogs:catalog-hive:compileJava *
     |    |         |    \--- :catalogs:catalog-hive:check
     |    |         |         +--- :catalogs:catalog-hive:spotlessCheck
     |    |         |         |    +--- :catalogs:catalog-hive:spotlessJavaCheck
     |    |         |         |    |    \--- :catalogs:catalog-hive:spotlessJava
     |    |         |         |    |         \--- :spotlessInternalRegisterDependencies *
     |    |         |         |    \--- :catalogs:catalog-hive:spotlessKotlinGradleCheck
     |    |         |         |         \--- :catalogs:catalog-hive:spotlessKotlinGradle
     |    |         |         |              \--- :spotlessInternalRegisterDependencies *
     |    |         |         \--- :catalogs:catalog-hive:test
     |    |         |              +--- :api:jar *
     |    |         |              +--- :common:jar *
     |    |         |              +--- :core:jar *
     |    |         |              +--- :meta:jar *
     |    |         |              +--- :catalogs:catalog-hive:classes *
     |    |         |              +--- :catalogs:catalog-hive:compileJava *
     |    |         |              +--- :catalogs:catalog-hive:compileTestJava
     |    |         |              |    +--- :api:jar *
     |    |         |              |    +--- :core:jar *
     |    |         |              |    +--- :catalogs:catalog-hive:classes *
     |    |         |              |    \--- :catalogs:catalog-hive:compileJava *
     |    |         |              \--- :catalogs:catalog-hive:testClasses
     |    |         |                   +--- :catalogs:catalog-hive:compileTestJava *
     |    |         |                   \--- :catalogs:catalog-hive:processTestResources
     |    |         \--- :catalogs:catalog-hive:copyDepends
     |    |              +--- :api:jar *
     |    |              +--- :common:jar *
     |    |              +--- :core:jar *
     |    |              \--- :meta:jar *
     |    +--- :catalogs:catalog-jdbc-mysql:copyLibAndConfig
     |    |    +--- :catalogs:catalog-jdbc-mysql:copyCatalogConfig
     |    |    \--- :catalogs:catalog-jdbc-mysql:copyCatalogLibs
     |    |         +--- :catalogs:catalog-jdbc-mysql:build
     |    |         |    +--- :catalogs:catalog-jdbc-mysql:assemble
     |    |         |    |    \--- :catalogs:catalog-jdbc-mysql:jar
     |    |         |    |         +--- :catalogs:catalog-jdbc-mysql:classes
     |    |         |    |         |    +--- :catalogs:catalog-jdbc-mysql:compileJava
     |    |         |    |         |    |    +--- :api:jar *
     |    |         |    |         |    |    +--- :common:jar *
     |    |         |    |         |    |    +--- :core:jar *
     |    |         |    |         |    |    \--- :catalogs:catalog-jdbc-common:jar
     |    |         |    |         |    |         +--- :catalogs:catalog-jdbc-common:classes
     |    |         |    |         |    |         |    +--- :catalogs:catalog-jdbc-common:compileJava
     |    |         |    |         |    |         |    |    +--- :api:jar *
     |    |         |    |         |    |         |    |    +--- :common:jar *
     |    |         |    |         |    |         |    |    \--- :core:jar *
     |    |         |    |         |    |         |    \--- :catalogs:catalog-jdbc-common:processResources
     |    |         |    |         |    |         \--- :catalogs:catalog-jdbc-common:compileJava *
     |    |         |    |         |    \--- :catalogs:catalog-jdbc-mysql:processResources
     |    |         |    |         \--- :catalogs:catalog-jdbc-mysql:compileJava *
     |    |         |    \--- :catalogs:catalog-jdbc-mysql:check
     |    |         |         +--- :catalogs:catalog-jdbc-mysql:spotlessCheck
     |    |         |         |    +--- :catalogs:catalog-jdbc-mysql:spotlessJavaCheck
     |    |         |         |    |    \--- :catalogs:catalog-jdbc-mysql:spotlessJava
     |    |         |         |    |         \--- :spotlessInternalRegisterDependencies *
     |    |         |         |    \--- :catalogs:catalog-jdbc-mysql:spotlessKotlinGradleCheck
     |    |         |         |         \--- :catalogs:catalog-jdbc-mysql:spotlessKotlinGradle
     |    |         |         |              \--- :spotlessInternalRegisterDependencies *
     |    |         |         \--- :catalogs:catalog-jdbc-mysql:test
     |    |         |              +--- :api:jar *
     |    |         |              +--- :common:jar *
     |    |         |              +--- :core:jar *
     |    |         |              +--- :meta:jar *
     |    |         |              +--- :catalogs:catalog-jdbc-common:jar *
     |    |         |              +--- :catalogs:catalog-jdbc-mysql:classes *
     |    |         |              +--- :catalogs:catalog-jdbc-mysql:compileJava *
     |    |         |              +--- :catalogs:catalog-jdbc-mysql:compileTestJava
     |    |         |              |    +--- :api:jar *
     |    |         |              |    +--- :common:jar *
     |    |         |              |    +--- :core:jar *
     |    |         |              |    +--- :catalogs:catalog-jdbc-common:jar *
     |    |         |              |    +--- :catalogs:catalog-jdbc-mysql:classes *
     |    |         |              |    \--- :catalogs:catalog-jdbc-mysql:compileJava *
     |    |         |              \--- :catalogs:catalog-jdbc-mysql:testClasses
     |    |         |                   +--- :catalogs:catalog-jdbc-mysql:compileTestJava *
     |    |         |                   \--- :catalogs:catalog-jdbc-mysql:processTestResources
     |    |         \--- :catalogs:catalog-jdbc-mysql:copyDepends
     |    |              +--- :api:jar *
     |    |              +--- :common:jar *
     |    |              +--- :core:jar *
     |    |              +--- :meta:jar *
     |    |              \--- :catalogs:catalog-jdbc-common:jar *
     |    +--- :catalogs:catalog-jdbc-postgresql:copyLibAndConfig
     |    |    +--- :catalogs:catalog-jdbc-postgresql:copyCatalogConfig
     |    |    \--- :catalogs:catalog-jdbc-postgresql:copyCatalogLibs
     |    |         +--- :catalogs:catalog-jdbc-postgresql:build
     |    |         |    +--- :catalogs:catalog-jdbc-postgresql:assemble
     |    |         |    |    \--- :catalogs:catalog-jdbc-postgresql:jar
     |    |         |    |         +--- :catalogs:catalog-jdbc-postgresql:classes
     |    |         |    |         |    +--- :catalogs:catalog-jdbc-postgresql:compileJava
     |    |         |    |         |    |    +--- :api:jar *
     |    |         |    |         |    |    +--- :common:jar *
     |    |         |    |         |    |    +--- :core:jar *
     |    |         |    |         |    |    \--- :catalogs:catalog-jdbc-common:jar *
     |    |         |    |         |    \--- :catalogs:catalog-jdbc-postgresql:processResources
     |    |         |    |         \--- :catalogs:catalog-jdbc-postgresql:compileJava *
     |    |         |    \--- :catalogs:catalog-jdbc-postgresql:check
     |    |         |         +--- :catalogs:catalog-jdbc-postgresql:spotlessCheck
     |    |         |         |    +--- :catalogs:catalog-jdbc-postgresql:spotlessJavaCheck
     |    |         |         |    |    \--- :catalogs:catalog-jdbc-postgresql:spotlessJava
     |    |         |         |    |         \--- :spotlessInternalRegisterDependencies *
     |    |         |         |    \--- :catalogs:catalog-jdbc-postgresql:spotlessKotlinGradleCheck
     |    |         |         |         \--- :catalogs:catalog-jdbc-postgresql:spotlessKotlinGradle
     |    |         |         |              \--- :spotlessInternalRegisterDependencies *
     |    |         |         \--- :catalogs:catalog-jdbc-postgresql:test
     |    |         |              +--- :api:jar *
     |    |         |              +--- :common:jar *
     |    |         |              +--- :core:jar *
     |    |         |              +--- :meta:jar *
     |    |         |              +--- :catalogs:catalog-jdbc-common:jar *
     |    |         |              +--- :catalogs:catalog-jdbc-postgresql:classes *
     |    |         |              +--- :catalogs:catalog-jdbc-postgresql:compileJava *
     |    |         |              +--- :catalogs:catalog-jdbc-postgresql:compileTestJava
     |    |         |              |    +--- :api:jar *
     |    |         |              |    +--- :common:jar *
     |    |         |              |    +--- :core:jar *
     |    |         |              |    +--- :catalogs:catalog-jdbc-common:jar *
     |    |         |              |    +--- :catalogs:catalog-jdbc-postgresql:classes *
     |    |         |              |    \--- :catalogs:catalog-jdbc-postgresql:compileJava *
     |    |         |              \--- :catalogs:catalog-jdbc-postgresql:testClasses
     |    |         |                   +--- :catalogs:catalog-jdbc-postgresql:compileTestJava *
     |    |         |                   \--- :catalogs:catalog-jdbc-postgresql:processTestResources
     |    |         \--- :catalogs:catalog-jdbc-postgresql:copyDepends
     |    |              +--- :api:jar *
     |    |              +--- :common:jar *
     |    |              +--- :core:jar *
     |    |              +--- :meta:jar *
     |    |              \--- :catalogs:catalog-jdbc-common:jar *
     |    \--- :catalogs:catalog-lakehouse-iceberg:copyLibAndConfig
     |         +--- :catalogs:catalog-lakehouse-iceberg:copyCatalogConfig
     |         \--- :catalogs:catalog-lakehouse-iceberg:copyCatalogLibs
     |              +--- :catalogs:catalog-lakehouse-iceberg:build
     |              |    +--- :catalogs:catalog-lakehouse-iceberg:assemble
     |              |    |    \--- :catalogs:catalog-lakehouse-iceberg:jar
     |              |    |         +--- :catalogs:catalog-lakehouse-iceberg:classes
     |              |    |         |    +--- :catalogs:catalog-lakehouse-iceberg:compileJava
     |              |    |         |    |    +--- :api:jar *
     |              |    |         |    |    +--- :common:jar *
     |              |    |         |    |    +--- :core:jar *
     |              |    |         |    |    \--- :server-common:jar
     |              |    |         |    |         +--- :server-common:classes
     |              |    |         |    |         |    +--- :server-common:compileJava
     |              |    |         |    |         |    |    +--- :api:jar *
     |              |    |         |    |         |    |    +--- :common:jar *
     |              |    |         |    |         |    |    \--- :core:jar *
     |              |    |         |    |         |    \--- :server-common:processResources
     |              |    |         |    |         \--- :server-common:compileJava *
     |              |    |         |    \--- :catalogs:catalog-lakehouse-iceberg:processResources
     |              |    |         \--- :catalogs:catalog-lakehouse-iceberg:compileJava *
     |              |    \--- :catalogs:catalog-lakehouse-iceberg:check
     |              |         +--- :catalogs:catalog-lakehouse-iceberg:spotlessCheck
     |              |         |    +--- :catalogs:catalog-lakehouse-iceberg:spotlessJavaCheck
     |              |         |    |    \--- :catalogs:catalog-lakehouse-iceberg:spotlessJava
     |              |         |    |         \--- :spotlessInternalRegisterDependencies *
     |              |         |    \--- :catalogs:catalog-lakehouse-iceberg:spotlessKotlinGradleCheck
     |              |         |         \--- :catalogs:catalog-lakehouse-iceberg:spotlessKotlinGradle
     |              |         |              \--- :spotlessInternalRegisterDependencies *
     |              |         \--- :catalogs:catalog-lakehouse-iceberg:test
     |              |              +--- :api:jar *
     |              |              +--- :common:jar *
     |              |              +--- :core:jar *
     |              |              +--- :meta:jar *
     |              |              +--- :server-common:jar *
     |              |              +--- :catalogs:catalog-lakehouse-iceberg:classes *
     |              |              +--- :catalogs:catalog-lakehouse-iceberg:compileJava *
     |              |              +--- :catalogs:catalog-lakehouse-iceberg:compileTestJava
     |              |              |    +--- :api:jar *
     |              |              |    +--- :common:jar *
     |              |              |    +--- :core:jar *
     |              |              |    +--- :server-common:jar *
     |              |              |    +--- :catalogs:catalog-lakehouse-iceberg:classes *
     |              |              |    \--- :catalogs:catalog-lakehouse-iceberg:compileJava *
     |              |              \--- :catalogs:catalog-lakehouse-iceberg:testClasses
     |              |                   +--- :catalogs:catalog-lakehouse-iceberg:compileTestJava *
     |              |                   \--- :catalogs:catalog-lakehouse-iceberg:processTestResources
     |              \--- :catalogs:catalog-lakehouse-iceberg:copyDepends
     |                   +--- :api:jar *
     |                   +--- :common:jar *
     |                   +--- :core:jar *
     |                   +--- :meta:jar *
     |                   \--- :server-common:jar *
     +--- :copySubprojectDependencies
     |    +--- :api:jar *
     |    +--- :common:jar *
     |    +--- :core:jar *
     |    +--- :meta:jar *
     |    \--- :server-common:jar *
     \--- :copySubprojectLib
          +--- :api:build
          |    +--- :api:assemble
          |    |    \--- :api:jar *
          |    +--- :api:check
          |    |    +--- :api:spotlessCheck
          |    |    |    +--- :api:spotlessJavaCheck
          |    |    |    |    \--- :api:spotlessJava
          |    |    |    |         \--- :spotlessInternalRegisterDependencies *
          |    |    |    \--- :api:spotlessKotlinGradleCheck
          |    |    |         \--- :api:spotlessKotlinGradle
          |    |    |              \--- :spotlessInternalRegisterDependencies *
          |    |    \--- :api:test
          |    |         +--- :api:classes *
          |    |         +--- :api:compileJava *
          |    |         +--- :api:compileTestJava
          |    |         |    +--- :api:classes *
          |    |         |    \--- :api:compileJava *
          |    |         \--- :api:testClasses
          |    |              +--- :api:compileTestJava *
          |    |              \--- :api:processTestResources
          |    \--- :api:javadoc
          |         +--- :api:classes *
          |         \--- :api:compileJava *
          +--- :common:build
          |    +--- :common:assemble
          |    |    \--- :common:jar *
          |    \--- :common:check
          |         +--- :common:spotlessCheck
          |         |    +--- :common:spotlessJavaCheck
          |         |    |    \--- :common:spotlessJava
          |         |    |         \--- :spotlessInternalRegisterDependencies *
          |         |    \--- :common:spotlessKotlinGradleCheck
          |         |         \--- :common:spotlessKotlinGradle
          |         |              \--- :spotlessInternalRegisterDependencies *
          |         \--- :common:test
          |              +--- :api:jar *
          |              +--- :common:classes *
          |              +--- :common:compileJava *
          |              +--- :common:compileTestJava
          |              |    +--- :api:jar *
          |              |    +--- :common:classes *
          |              |    \--- :common:compileJava *
          |              \--- :common:testClasses
          |                   +--- :common:compileTestJava *
          |                   \--- :common:processTestResources
          +--- :core:build
          |    +--- :core:assemble
          |    |    \--- :core:jar *
          |    \--- :core:check
          |         +--- :core:spotlessCheck
          |         |    +--- :core:spotlessJavaCheck
          |         |    |    \--- :core:spotlessJava
          |         |    |         \--- :spotlessInternalRegisterDependencies *
          |         |    \--- :core:spotlessKotlinGradleCheck
          |         |         \--- :core:spotlessKotlinGradle
          |         |              \--- :spotlessInternalRegisterDependencies *
          |         \--- :core:test
          |              +--- :api:jar *
          |              +--- :common:jar *
          |              +--- :core:classes *
          |              +--- :core:compileJava *
          |              +--- :core:compileTestJava
          |              |    +--- :api:jar *
          |              |    +--- :common:jar *
          |              |    +--- :core:classes *
          |              |    +--- :core:compileJava *
          |              |    \--- :meta:jar *
          |              +--- :core:testClasses
          |              |    +--- :core:compileTestJava *
          |              |    \--- :core:processTestResources
          |              \--- :meta:jar *
          +--- :docs:build
          |    +--- :docs:assemble
          |    |    \--- :docs:jar
          |    |         +--- :docs:classes
          |    |         |    +--- :docs:compileJava
          |    |         |    \--- :docs:processResources
          |    |         \--- :docs:compileJava *
          |    +--- :docs:check
          |    |    +--- :docs:spotlessCheck
          |    |    |    +--- :docs:spotlessJavaCheck
          |    |    |    |    \--- :docs:spotlessJava
          |    |    |    |         \--- :spotlessInternalRegisterDependencies *
          |    |    |    \--- :docs:spotlessKotlinGradleCheck
          |    |    |         \--- :docs:spotlessKotlinGradle
          |    |    |              \--- :spotlessInternalRegisterDependencies *
          |    |    \--- :docs:test
          |    |         +--- :docs:classes *
          |    |         +--- :docs:compileJava *
          |    |         +--- :docs:compileTestJava
          |    |         |    +--- :docs:classes *
          |    |         |    \--- :docs:compileJava *
          |    |         \--- :docs:testClasses
          |    |              +--- :docs:compileTestJava *
          |    |              \--- :docs:processTestResources
          |    \--- :docs:lintOpenAPI
          |         \--- :docs:npmSetup
          |              \--- :docs:nodeSetup
          +--- :meta:build
          |    +--- :meta:assemble
          |    |    \--- :meta:jar *
          |    \--- :meta:check
          |         +--- :meta:spotlessCheck
          |         |    +--- :meta:spotlessJavaCheck
          |         |    |    \--- :meta:spotlessJava
          |         |    |         +--- :spotlessInternalRegisterDependencies *
          |         |    |         +--- :meta:generateProto *
          |         |    |         \--- :meta:generateTestProto
          |         |    |              +--- :meta:extractIncludeProto *
          |         |    |              +--- :meta:extractIncludeTestProto
          |         |    |              +--- :meta:extractProto *
          |         |    |              \--- :meta:extractTestProto
          |         |    \--- :meta:spotlessKotlinGradleCheck
          |         |         \--- :meta:spotlessKotlinGradle
          |         |              \--- :spotlessInternalRegisterDependencies *
          |         \--- :meta:test
          |              +--- :meta:classes *
          |              +--- :meta:compileJava *
          |              +--- :meta:compileTestJava
          |              |    +--- :meta:classes *
          |              |    +--- :meta:compileJava *
          |              |    \--- :meta:generateTestProto *
          |              \--- :meta:testClasses
          |                   +--- :meta:compileTestJava *
          |                   \--- :meta:processTestResources
          |                        \--- :meta:extractTestProto *
          +--- :server:build
          |    +--- :server:assemble
          |    |    \--- :server:jar
          |    |         +--- :server:classes
          |    |         |    +--- :server:compileJava
          |    |         |    |    +--- :api:jar *
          |    |         |    |    +--- :common:jar *
          |    |         |    |    +--- :core:jar *
          |    |         |    |    \--- :server-common:jar *
          |    |         |    \--- :server:processResources
          |    |         \--- :server:compileJava *
          |    \--- :server:check
          |         +--- :server:spotlessCheck
          |         |    +--- :server:spotlessJavaCheck
          |         |    |    \--- :server:spotlessJava
          |         |    |         \--- :spotlessInternalRegisterDependencies *
          |         |    \--- :server:spotlessKotlinGradleCheck
          |         |         \--- :server:spotlessKotlinGradle
          |         |              \--- :spotlessInternalRegisterDependencies *
          |         \--- :server:test
          |              +--- :api:jar *
          |              +--- :common:jar *
          |              +--- :core:jar *
          |              +--- :meta:jar *
          |              +--- :server:classes *
          |              +--- :server:compileJava *
          |              +--- :server:compileTestJava
          |              |    +--- :api:jar *
          |              |    +--- :common:jar *
          |              |    +--- :core:jar *
          |              |    +--- :server:classes *
          |              |    +--- :server:compileJava *
          |              |    \--- :server-common:jar *
          |              +--- :server:testClasses
          |              |    +--- :server:compileTestJava *
          |              |    \--- :server:processTestResources
          |              \--- :server-common:jar *
          +--- :server-common:build
          |    +--- :server-common:assemble
          |    |    \--- :server-common:jar *
          |    \--- :server-common:check
          |         +--- :server-common:spotlessCheck
          |         |    +--- :server-common:spotlessJavaCheck
          |         |    |    \--- :server-common:spotlessJava
          |         |    |         \--- :spotlessInternalRegisterDependencies *
          |         |    \--- :server-common:spotlessKotlinGradleCheck
          |         |         \--- :server-common:spotlessKotlinGradle
          |         |              \--- :spotlessInternalRegisterDependencies *
          |         \--- :server-common:test
          |              +--- :api:jar *
          |              +--- :common:jar *
          |              +--- :core:jar *
          |              +--- :meta:jar *
          |              +--- :server-common:classes *
          |              +--- :server-common:compileJava *
          |              +--- :server-common:compileTestJava
          |              |    +--- :api:jar *
          |              |    +--- :common:jar *
          |              |    +--- :core:jar *
          |              |    +--- :server-common:classes *
          |              |    \--- :server-common:compileJava *
          |              \--- :server-common:testClasses
          |                   +--- :server-common:compileTestJava *
          |                   \--- :server-common:processTestResources
          \--- :web:build
               +--- :web:assemble
               |    \--- :web:war
               |         +--- :web:classes
               |         |    +--- :web:compileJava
               |         |    \--- :web:processResources
               |         \--- :web:compileJava *
               +--- :web:buildWar
               |    +--- :web:classes *
               |    +--- :web:compileJava *
               |    \--- :web:webpack
               |         +--- :web:lintCheck
               |         |    +--- :web:yarnInstall
               |         |    |    \--- :web:yarnSetup
               |         |    |         \--- :web:nodeSetup
               |         |    \--- :web:yarnSetup *
               |         +--- :web:prettierCheck
               |         |    +--- :web:yarnInstall *
               |         |    \--- :web:yarnSetup *
               |         \--- :web:yarnSetup *
               \--- :web:check
                    +--- :web:spotlessCheck
                    |    +--- :web:spotlessJavaCheck
                    |    |    \--- :web:spotlessJava
                    |    |         \--- :spotlessInternalRegisterDependencies *
                    |    \--- :web:spotlessKotlinGradleCheck
                    |         \--- :web:spotlessKotlinGradle
                    |              \--- :spotlessInternalRegisterDependencies *
                    \--- :web:test
                         +--- :web:classes *
                         +--- :web:compileJava *
                         +--- :web:compileTestJava
                         |    +--- :web:classes *
                         |    \--- :web:compileJava *
                         \--- :web:testClasses
                              +--- :web:compileTestJava *
                              \--- :web:processTestResources
```

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -42,6 +42,7 @@ plugins {
   alias(libs.plugins.rat)
   id("com.github.jk1.dependency-license-report") version "2.5"
   id("org.cyclonedx.bom") version "1.5.0" // Newer version fail due to our setup
+  id("com.dorongold.task-tree") version "2.1.1"
```
Comment (jerryshao) [inline build.gradle.kts:45]: Please use the gradle version file to manage the plugin version.

Comment (jerryshao) [reply build.gradle.kts:45]: Also please update the above lines, seems there're not managed by version file.

Comment (jerryshao) [reply build.gradle.kts]: > Also please update the above lines, seems there're not managed by version file.

You'd also change the above lines.

```
  id("com.github.jk1.dependency-license-report") version "2.5"
  id("org.cyclonedx.bom") version "1.5.0" // Newer version fail due to our setup
```

Comment (yuqi1129) [reply build.gradle.kts]: OK, I will change them all.

Comment (yuqi1129) [reply build.gradle.kts]: Resolved

**Review by jerryshao — APPROVED** [review]

**Pull request merged**